flink-runtime/src/main/java/org/apache/flink/runtime/metrics/groups/TaskIOMetricGroup.java

```diff
@@ -67,10 +70,12 @@ public class TaskIOMetricGroup extends ProxyMetricGroup<TaskMetricGroup> {
     private final Meter mailboxThroughput;
     private final Histogram mailboxLatency;
     private final SizeGauge mailboxSize;
+    private final Gauge<Long> initializingTime;
```

Comment:
Thanks for the suggestion, I originally wanted to keep its semantics consistent with `initializingTime` at the job level, `initializingTime` will return 0 when the task is not in initialization, and it will decrease to 0 the moment when the task becomes running.  I will change it to `initializationDuration`, and let the duration stay after the task becomes running.

Reply:
Thanks! 
I see in the latest update that the type of the field is still gauge and not counter. Is it intentional?

Reply:
@rkhachatryan  Because a counter only provides `dec()/inc()` interfaces,  and `initializationDuration  = System.currentTimeMillis() - taskInitializeTime`, we don't know how much this sampling has increased compared to the previous sampling, I think gauge is more suitable for this scenario.

Reply:
I see. One way to still make it a counter would be to implement `View` + `Counter` which would monotinically increase the value in `update()`. That would keep semantics clearer.
WDYT?

Reply:
Sorry for the late reply, I changed the type of `initializationDuration` to Counter.

And I found that the semantic of counter in flink is not the same as the counter in Prometheus. The counter in flink provides a `dec()` interface, so the value of the counter may be not monotonically increasing as Prometheus.